### PR TITLE
Reduce the number of calls to get a singleton instance

### PR DESCRIPTION
**Pull request by ihhub**
Just a simple code cleanup to reduce the amount of useless work by CPU.

### CODE REVIEW

File: src/fheroes2/dialog/dialog_selectitems.cpp [hunk 1]
```diff
@@ -126,16 +126,18 @@ class SelectEnumMonster : public SelectEnum
...
         if ( current ) {
             fheroes2::Text text( mons.GetName(), fheroes2::FontType::normalYellow() );
-            text.draw( dstx + 50, dsty + 10, fheroes2::Display::instance() );
+            text.draw( dstx + 50, dsty + 10, display );
         }
         else {
             fheroes2::Text text( mons.GetName(), fheroes2::FontType() );
-            text.draw( dstx + 50, dsty + 10, fheroes2::Display::instance() );
+            text.draw( dstx + 50, dsty + 10, display );
         }
```
Comment (Districh-ru) [inline src/fheroes2/dialog/dialog_selectitems.cpp:141]: We can reduce code duplication a bit here. (And explicitly indicate the `normalWhite()` font type.) What do you think?
```cpp
        const fheroes2::Text text( mons.GetName(), current ? fheroes2::FontType::normalYellow() : fheroes2::FontType::normalWhite() );
        text.draw( dstx + 50, dsty + 10, display );

```

And to finally clean-up this function we can make **const** line 129: `const Monster mons( index );`. :)

Also the same clean-up, including introducing `fheroes2::Display & display = fheroes2::Display::instance();`, can be done to `SelectEnumHeroes::RedrawItem()` (lines 166-181) and `SelectEnumArtifact::RedrawItem()` (lines 197-214).

File: src/fheroes2/dialog/dialog_selectitems.cpp [hunk 2]
```diff
@@ -233,16 +235,18 @@ class SelectEnumSpell : public SelectEnum
...
         if ( current ) {
             fheroes2::Text text( spell.GetName(), fheroes2::FontType::normalYellow() );
-            text.draw( dstx + 80, dsty + 10, fheroes2::Display::instance() );
+            text.draw( dstx + 80, dsty + 10, display );
         }
         else {
             fheroes2::Text text( spell.GetName(), fheroes2::FontType() );
-            text.draw( dstx + 80, dsty + 10, fheroes2::Display::instance() );
+            text.draw( dstx + 80, dsty + 10, display );
         }
```
Comment (Districh-ru) [inline src/fheroes2/dialog/dialog_selectitems.cpp:250]: The same comment as above.
```cpp
        const fheroes2::Text text( spell.GetName(), current ? fheroes2::FontType::normalYellow() : fheroes2::FontType::normalWhite() );
        text.draw( dstx + 80, dsty + 10, display );
```

File: src/fheroes2/dialog/dialog_selectitems.cpp [hunk 3]
```diff
@@ -266,17 +270,19 @@ class SelectEnumSecSkill : public SelectEnum
...
         if ( current ) {
             fheroes2::Text text( str, fheroes2::FontType::normalYellow() );
-            text.draw( dstx + 50, dsty + 10, fheroes2::Display::instance() );
+            text.draw( dstx + 50, dsty + 10, display );
         }
         else {
             fheroes2::Text text( str, fheroes2::FontType() );
-            text.draw( dstx + 50, dsty + 10, fheroes2::Display::instance() );
+            text.draw( dstx + 50, dsty + 10, display );
         }
```
Comment (Districh-ru) [inline src/fheroes2/dialog/dialog_selectitems.cpp:286]: The same comment as above.
```cpp
        const fheroes2::Text text( skill.GetName(), current ? fheroes2::FontType::normalYellow() : fheroes2::FontType::normalWhite() );
        text.draw( dstx + 50, dsty + 10, display );
```

**Review by Districh-ru — CHANGES_REQUESTED** [review]
Hi @ihhub. I left a code clean-up suggestion. Could you take a look in your spare time.

File: src/fheroes2/dialog/dialog_selectitems.cpp [hunk 4]
```diff
@@ -167,17 +163,12 @@ class SelectEnumHeroes : public SelectEnum
     {
         const fheroes2::Sprite & port = Heroes::GetPortrait( index, PORT_SMALL );
 
-        if ( !port.empty() )
+        if ( !port.empty() ) {
             fheroes2::Blit( port, fheroes2::Display::instance(), dstx + 5, dsty + 3 );
```
Comment (Districh-ru) [inline src/fheroes2/dialog/dialog_selectitems.cpp:166]: As every hero in non-corrupted original assets has a portrait this condition will be true and `fheroes2::Display::instance()` is called twice: here and in line 171.
So we can reduce the number of these system calls.

**Review by Districh-ru — CHANGES_REQUESTED** [review]
Hi @ihhub. I left one suggestion. Could you take a look when you have time.

File: src/fheroes2/dialog/dialog_selectitems.cpp (hunk 4 above)
Comment (ihhub) [reply src/fheroes2/dialog/dialog_selectitems.cpp]: Fixed.

**Review by Districh-ru — APPROVED** [review]

**Pull request merged**